### DISCUSSION TITLE
Github user wrongly assumed my code doesn't work and blocked me

### DISCUSSION THREAD

**Discussion by mruncreative** [community/community#146808] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

Github user wrongly assumed my code doesn't work, closed my PR and blocked me.
https://github.com/dessalines/thumb-key/pull/1186
I can see that he doesn't like that I didn't test it on real hardware right away but it wasn't much different than adding a keyboard layout which is done in code too in his application.
I just wrote the code using logic and compiling snippets in an online compiler.

I've compiled and tested it and it works fine.
[app-release.zip](https://github.com/user-attachments/files/18094248/app-release.zip)
There was no reason to throw a fit.

The problem is I can't tell him that I've compiled and tested it (Figuring out how to build it myself took indeed hours) because he doesn't provide any contact information aside from some Mastodon of which I'm not sure he's active on and his own clone of Mastodon on which he is the administrator and you can't even register without "applying for registration" which he then denies.

How do I get my code merged?
There is at least one other user who liked my previous addition.
https://github.com/dessalines/thumb-key/issues/1181

**Comment by karasowles** [comment]
Hi - it looks like you made a contribution to an open source project, the code was untested, and the the contribution was declined. This is pretty standard; Open Source Projects won't merge untested code, and maintainers rarely have the time to test code for contributors. 

I'd recommend moving on rather than trying to continue to reach out the project maintainer. They've made it clear they don't want to engage.

This guide talks some about expected behavior when contributing to open source: https://opensource.guide/how-to-contribute/
This post also has some good advice: https://dev.to/github/how-to-be-a-good-open-source-contributor-during-hacktoberfest-and-beyond-cdi

**Comment by mruncreative** [comment]
But the code IS tested now and it works. It worked all along. Maintainers should also test code before merging either way. He also accepted a contribution from me before.
Programming isn't different from mathematics in that you can formally verify it if it isn't too complicated.

**Comment by Roshanjossey** [comment]
Hi @mruncreative, I could provide some insights from a maintainer's perspective.

### The human side of open source.

> Programming isn't different from mathematics

Mostly agree.

Open source is more human though. That might be the perspective you're missing.
When a maintainer adopts a code of conduct or writes a CONTRIBUTING.md, they're also setting boundaries.

When you push those boundaries, the maintainer is right to give their well-being more priority and stop engaging in the discussion.
https://mikemcquaid.com/open-source-maintainers-owe-you-nothing/

So, when you engage in open source, think about the maintainers' perspective too.
